src/Security/Authenticator.php

```diff
@@ -117,9 +117,10 @@ public static function register_authenticator($authenticator)
             return false;
         }
 
-        if (in_array($authenticator, self::$authenticators) == false) {
-            if (call_user_func(array($authenticator, 'on_register')) === true) {
-                array_push(self::$authenticators, $authenticator);
+        $authenticators = self::config()->get('authenticators');
+        if (in_array($authenticator, $authenticators) == false) {
+            if ($authenticator::on_register() === true) {
+                self::config()->merge('authenticators', [$authenticator]);
```

Comment:
Let's deprecate unregister, and move to config-only registration? That way we can add a cache on top of config rather than through it.